app/views/facility_orders/index.html.haml

```diff
@@ -28,20 +28,18 @@
       -## THE DETAIL ROWS
       %tbody
         - @order_details.each do |od|
-          - order = od.order
           %tr
             %td.centered= check_box_tag("order_detail_ids[]", od.id, false, {:class => 'toggle', :id => nil })
-            %td.centered= link_to od.order_id, facility_order_path(current_facility, order)
+            %td.centered= link_to od.order_id, facility_order_path(current_facility, od.order)
             %td.centered= link_to od.id, manage_order_detail_path(od), :class => 'manage-order-detail'
-            %td=h human_datetime(order.ordered_at)
-            %td= order.user.full_name
-            %td.centered= od.quantity
+            %td= human_datetime(od.order.ordered_at)
+            %td= od.order.user.full_name
+            %td.centered= OrderDetailPresenter.new(od).wrapped_quantity
             = render :partial => 'shared/order_detail_cell', :locals => { :od => od }
             %td= od.assigned_user
             %td= od.order_status
             %td.currency
-              - od.send(:extend, PriceDisplayment)
-              = od.wrapped_total
+              =  OrderDetailPresenter.new(od).wrapped_total
```

Comment:
I don't think so. Since this is using a presenter, I would actually consider it view code. We're in a loop over many order details, so it would be tricky to get this into a controller. There's also a problem Meara and I ran into yesterday where cancan doesn't like presenters/decorators/delegators. `can?(:something, decorated_object)` will return false while the `can?(:something, object)` will return true. So we do want to keep the objects unwrapped where we don't need them.